dev-python/typer/typer-0.15.3.ebuild

```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{11..13} )
+DISTUTILS_USE_PEP517=pdm-backend
+
+inherit distutils-r1 shell-completion pypi
+
+DESCRIPTION="Build great CLIs. Easy to code. Based on Python type hints."
+HOMEPAGE="
+	https://typer.tiangolo.com/
+	https://github.com/tiangolo/typer
+	https://pypi.org/project/typer/
+"
+SRC_URI="$(pypi_sdist_url "${PN}" "${PV}")"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64"
+
+IUSE="+rich shell-detection test"
+
+BDEPEND="
+		dev-python/shellingham[${PYTHON_USEDEP}]
+"
+RDEPEND="
+	dev-python/click[${PYTHON_USEDEP}]
+	dev-python/typing-extensions[${PYTHON_USEDEP}]
+	rich? ( dev-python/rich[${PYTHON_USEDEP}] )
+	shell-detection? ( dev-python/shellingham[${PYTHON_USEDEP}] )
+"
+
+python_prepare_all() {
+	# clean up coverage
+	sed -i \
+		-e 's/"-m", "coverage", "run",//' \
+		-e 's/-m coverage run//' \
+		-e '/"-m",/ {
+			N
+			N
+			s/"-m",\n.*"coverage",\n.*"run",//
+		}' $(find tests -name "test_*.py") || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	# See scripts/tests.sh
+	local -x TERMINAL_WIDTH=3000 _TYPER_FORCE_DISABLE_TERMINAL=1  _TYPER_RUN_INSTALL_COMPLETION_TESTS=1
+
+	epytest
+}
+distutils_enable_tests pytest
+
+src_test() {
+	export TERMINAL_WIDTH=3000
+	export _TYPER_FORCE_DISABLE_TERMINAL=1
+
+	distutils-r1_src_test "${@}"
+}
+
+src_compile() {
+	distutils-r1_src_compile "${@}"
+
+	local shell
+	for shell in bash zsh fish; do
+		typer_gencomp ${shell}
+	done
+}
+
+typer_get_comp() {
+	local COMPLETIONSDIR="${WORKDIR}/comp"
+	local shell="$1"
+
+	case "${shell}" in
+		bash) echo "${COMPLETIONSDIR}/${PN}" ;;
+		zsh)  echo "${COMPLETIONSDIR}/_${PN}" ;;
+		fish) echo "${COMPLETIONSDIR}/${PN}.fish" ;;
+		*)    die "unknown shell: ${shell}" ;;
+	esac
+}
```

Comment:
I'm pretty sure you'll be able to remove most of this nonsense if you just defer generation until `python_install`, and then install them immediately.